internal/local/app_test.go

```diff
@@ -673,87 +698,39 @@ exports = function({ query }) {
-	Hosting: map[string]interface{}{
-		"enabled":            true,
-		"app_default_domain": "full-tkdcx.stitch-statichosting-dev.baas-dev.10gen.cc",
-	},
-	CustomUserDataConfig: map[string]interface{}{
-		"enabled":            true,
-		"mongo_service_name": "mongodb-atlas",
-		"database_name":      "test",
-		"collection_name":    "coll3",
-		"user_id_field":      "xref",
-	},
-	Sync: map[string]interface{}{"development_mode_enabled": false},
-}}
-
-var appData20200603Local = AppDataV1{AppStructureV1{
-	ConfigVersion:   realm.AppConfigVersion20200603,
-	Name:            "20200603-local",
-	Location:        realm.LocationVirginia,
-	DeploymentModel: realm.DeploymentModelGlobal,
-	Security:        map[string]interface{}{"allowed_request_origins": []interface{}{"http://localhost:8080"}},
-	Hosting: map[string]interface{}{
-		"enabled":            true,
-		"app_default_domain": "full-tkdcx.stitch-statichosting-dev.baas-dev.10gen.cc",
-	},
-	CustomUserDataConfig: map[string]interface{}{
-		"enabled":            true,
-		"mongo_service_name": "mongodb-atlas",
-		"database_name":      "test",
-		"collection_name":    "coll3",
-		"user_id_field":      "xref",
-	},
-	Sync: map[string]interface{}{"development_mode_enabled": false},
-}}
-
-var appData20200603Remote = AppDataV1{AppStructureV1{
-	ConfigVersion:   realm.AppConfigVersion20200603,
-	ID:              "20200603-remote-abcde",
-	Name:            "20200603-remote",
-	Location:        realm.LocationVirginia,
-	DeploymentModel: realm.DeploymentModelGlobal,
-	Security:        map[string]interface{}{"allowed_request_origins": []interface{}{"http://localhost:8080"}},
-	Hosting: map[string]interface{}{
-		"enabled":            true,
-		"app_default_domain": "full-tkdcx.stitch-statichosting-dev.baas-dev.10gen.cc",
-	},
-	CustomUserDataConfig: map[string]interface{}{
-		"enabled":            true,
-		"mongo_service_name": "mongodb-atlas",
-		"database_name":      "test",
-		"collection_name":    "coll3",
-		"user_id_field":      "xref",
-	},
-	Sync: map[string]interface{}{"development_mode_enabled": false},
-}}
+func createAppDataV1(version realm.AppConfigVersion, name string) AppDataV1 {
+	app := AppDataV1{AppStructureV1{
+		ConfigVersion:   version,
+		Name:            version.String() + "-" + name,
+		Location:        realm.LocationVirginia,
+		DeploymentModel: realm.DeploymentModelGlobal,
+		Security:        map[string]interface{}{"allowed_request_origins": []interface{}{"http://localhost:8080"}},
+		Hosting: map[string]interface{}{
+			"enabled":            true,
+			"app_default_domain": "full-tkdcx.stitch-statichosting-dev.baas-dev.10gen.cc",
+		},
+		CustomUserDataConfig: map[string]interface{}{
+			"enabled":            true,
+			"mongo_service_name": "mongodb-atlas",
+			"database_name":      "test",
+			"collection_name":    "coll3",
+			"user_id_field":      "xref",
+		},
+		Sync: map[string]interface{}{"development_mode_enabled": false},
+	}}
+
+	switch name {
+	case "remote":
```

Comment:
I don't _love_ having logic contained in this helper that requires knowledge of the various `name` inputs it could be passed.  (i also won't die on any hills here as it is just a test helper... 😬 )

one thought is to push this type of logic downstream of whatever's calling `createAppDataV1`... that has the unfortunate side effect for us of forcing a change in our test case fields.  basically the `AppData` fields need to become functional (e.g. `func() AppData` (not ideal!))

so let's consider the opposite approach: embracing the 6 different (but very similar) vars (with a few extra common `var`s, it becomes somewhat less crazy):

```go
var allowedRequestOrigins = []interface{}{...}

var appSecurity = map[string]interface{}{"allowed_request_origins": allowedRequestOrigins}

// other vars for the various re-used components

var appData20180301Local = AppDataV1{AppStructureV1{
  ConfigVersion:   realm.AppConfigVersion20180301,
  Name: "20180301-local",
  Location:        realm.LocationVirginia,
  DeploymentModel: realm.DeploymentModelGlobal,
  Security: appSecurity,
  Hosting: appHosting,
  CustomUserDataConfig: appCustomUserDataConfig,
  Sync: appSync,
}}
```